lib/portlayer/storage/vsphere/image.go

```diff
@@ -283,6 +244,119 @@ func (v *ImageStore) WriteImage(ctx context.Context, parent *portlayer.Image, ID
 	return newImage, nil
 }
 
+// Create a temporary directory, create a vmdk in this directory, attach/mount
+// the disk, unpack the tar, check the checksum.  If the data doesn't match the
+// expected checksum, abort by nuking the tempdir.  If everything matches, move
+// the tmpdir to the expected location (with the vmdk inside it).  The unwind
+// path is a bit convoluted here;  we need to clean up on the way out in the
+// error case (using the tmpdir).
+func (v *ImageStore) writeImage(ctx context.Context, storeName, parentID, ID,
+	sum string, r io.Reader) error {
+
+	// Create the image directory in the store.
+	tmpImageDir := v.imageDirPath(storeName, ID+".inprogress")
```

Comment:
I think docker is blocking other callers saying download in progress, not sure how...